GoogleForCreators/web-stories-wp
# Dashboard: Break apart my stories functionality (refactor)

## Summary
Last sprint when we added the filler page for saved templates we broke apart the view into header and content and found it organizes things quite well. Following up on that with an update to the My Stories view to get them structured the same. 

## Relevant Technical Choices
- Split Header and Content in MyStories into their own nested files to organize tests and storybook by component instead of page. 
- Added providers to preview for storybook dashboard to remove font errors 
- Move providers needed for stories to load to the Content component, we don't need to wrap the header in those 
- Grouped all the story actions as an object when passing to Content just because it feels more organized 
- I had to wrap the Content storybook examples in a container because we are using wordpress styles within the Layout provider that negatively impact rendering a Layout related item without a container! It's pretty obvious in storybook and should be fine for the purpose of stress testing the Content as a component.


## To-do
- Move Saved Templates content and header into subdirectories to match.
- More unit tests (need to move on from this refactor to work on bugs) 

## User-facing changes
- No user facing changes outside of storybook, which now has[ Dashboard/Components/myStories](http://localhost:55135/?path=/story/dashboard-components-mystories-header--default)

![Screen Shot 2020-05-20 at 9 48 05 AM](https://user-images.githubusercontent.com/10720454/82475051-c0d69080-9a80-11ea-97da-f8ec48c2ec3b.png)
<img width="1012" alt="Screen Shot 2020-05-20 at 12 22 00 PM" src="https://user-images.githubusercontent.com/10720454/82488353-968ece00-9a94-11ea-91c4-49f6598a31f9.png">


## Testing Instructions
- Ensure my stories page functionality and presentation is the same, no regressions
- See states of my stories header and content in storybook 

---



Fixes #
